### ISSUE TITLE
Create pretrained features baseline for portion of Figure 2

### ISSUE THREAD

**Issue by shntnu** [jump-cellpainting/2021_Chandrasekaran_submitted#20]
Let's create pretrained features baseline for portion of Figure 2

![image](https://user-images.githubusercontent.com/1210428/135119503-b254c309-e218-48c3-8f6c-99ca0acbea21.png)

**Comment by shntnu** [comment]
| Barcode    | Description                                     | Number_of_images |
| ---------- | ----------------------------------------------- | ---------------- |
| BR00118039 | U2OS 96-hour ORF Plate 1                        | 27648            |
| BR00118040 | U2OS 96-hour ORF Plate 2                        | 27648            |
| BR00117022 | U2OS 48-hour ORF Plate 1                        | 27648            |
| BR00117023 | U2OS 48-hour ORF Plate 2                        | 27648            |
| BR00118045 | U2OS 96-hour CRISPR Plate 1                     | 27648            |
| BR00118046 | U2OS 96-hour CRISPR Plate 2                     | 27640            |
| BR00118047 | U2OS 96-hour CRISPR Plate 3                     | 27648            |
| BR00118048 | U2OS 96-hour CRISPR Plate 4                     | 27648            |
| BR00116997 | U2OS 144-hour CRISPR Plate 1                    | 27648            |
| BR00116998 | U2OS 144-hour CRISPR Plate 2                    | 27648            |
| BR00116999 | U2OS 144-hour CRISPR Plate 3                    | 27648            |
| BR00116996 | U2OS 144-hour CRISPR Plate 4                    | 27648            |
| BR00116995 | U2OS 24-hour Compound Plate 1                   | 27648            |
| BR00117024 | U2OS 24-hour Compound Plate 2                   | 27648            |
| BR00117025 | U2OS 24-hour Compound Plate 3                   | 27648            |
| BR00117026 | U2OS 24-hour Compound Plate 4                   | 27648            |
| BR00117012 | U2OS 48-hour Compound Plate 1                   | 27648            |
| BR00117013 | U2OS 48-hour Compound Plate 2                   | 27648            |
| BR00117010 | U2OS 48-hour Compound Plate 3                   | 27648            |
| BR00117011 | U2OS 48-hour Compound Plate 4                   | 27648            |

**Comment by shntnu** [comment]
The plates  are at
 aws s3 ls --profile jump s3://jump-cellpainting/projects/2019_07_11_JUMP-CP-pilots/2020_11_04_CPJUMP1/images/
but it's quite likely we will need to initiate a restore

**Comment by shntnu** [comment]
@niranjchandrasekaran 
@johnarevalo  will provide 12 .csv.gz files as output:

```
BR00118039.csv.gz
BR00118040.csv.gz
BR00116997.csv.gz
BR00116998.csv.gz
BR00116999.csv.gz
BR00116996.csv.gz
BR00117012.csv.gz
BR00117013.csv.gz
BR00117010.csv.gz
BR00117011.csv.gz
```

They will have 
- one row per well
- 2 Metadata columns – Metadata_Plate and Metadata_Well
- several feature columns – not sure what their prefixes 

Would that be sufficient for you to produce the figure @niranjchandrasekaran ?

**Comment by niranjchandrasekaran** [comment]
Having `Metadata_broad_sample` will be helpful. But if it is not straightforward to do that, I will do the mapping myself.

> except that the feature names will be different (and not have a Cells_ etc. prefix)
That should be ok.

**Comment by shntnu** [comment]
> But if it is not straightforward to do that, I will do the mapping myself.

That will be much harder to do 

essentially the input will be identical to files like https://github.com/jump-cellpainting/neurips-cpjump1/blob/main/profiles/2020_11_04_CPJUMP1/BR00117012/BR00117012.csv.gz

so I think you can start at 
https://github.com/jump-cellpainting/profiling-recipe/blob/a9f01b359818737cd385fb6e0a94a170827d6fa7/profiles/profile.py#L102
right?

**Comment by niranjchandrasekaran** [comment]
I see... I can annotate the profiles. How about normalization and feature selection? Should we be doing sphering and feature selection like Michael has been doing with the LINCS data?

**Comment by shntnu** [comment]
> How about normalization and feature selection? Should we be doing sphering and feature selection like Michael has been doing with the LINCS data?

No feature selection. We would ideally spherize the data across all replicates of a plate map (in this case) but it depends on how much time you have @niranjchandrasekaran

**Comment by johnarevalo** [comment]
This folder contains the features extracted by efficientnet:

https://drive.google.com/drive/folders/17_FucoLbH4_yGl59CHJ24yJHR4RKeHoO

**Comment by shntnu** [comment]
This amazing @johnarevalo !!

**Comment by niranjchandrasekaran** [comment]
@shntnu Preliminary results - I annotated the plates, spherized them (one platemap at a time).

| Description      | Perturbation   | time   | Cell   |   Percent_Replicating |
|:-----------------|:---------------|:-------|:-------|----------------------:|
| ORF_U2OS_96      | ORF            | long   | U2OS   |                  41.9 |
| CRISPR_U2OS_144  | CRISPR         | long   | U2OS   |                  30.8 |
| Compound_U2OS_48 | Compound       | long   | U2OS   |                  59.5 |

![Screen Shot 2021-09-29 at 12 49 14 PM](https://user-images.githubusercontent.com/19695854/135313590-8708cc2f-2e9e-4eaf-bf35-eeab5c1df31b.png)

**Comment by shntnu** [comment]
Great! I the final revised version of the paper, we should also report the well position effect (=false discovery). I bet this version of the data (efn + spherized) will do better overall once we factor that in. We are all set for now – wow!

**Comment by shntnu** [comment]
@johnarevalo to wrap this up (from your end)

Could you please do the following?

- Document where the processed data was stored in S3
- Document bash commands that you ran to process the data e.g. https://github.com/broadinstitute/neuronal-cell-painting/blob/master/1.run-workflows/generate_profiles.sh. It does not need to be as detailed in terms of comments but should, ideally, be exactly the sequence of commands you ran on bash to get the processing done, which might include calls like `jupyter nbconvert --execute <notebook>`. This could be harder than I'm thinking, in which case LMK.

You can document both things right here in comments, to keep things simple

**Comment by tahereT** [comment]
@johnarevalo, how can I download the pretrained model ("efficientnetb0_notop.h5")?

**Comment by johnarevalo** [comment]
Hi @tahereT. Note that deepprofiler is in development stage, things may change without notice.

1. by `setting profile: checkpoint` to `"None"` (as string), deep profiler automatically downloads the pretrained weights.
2. The `metadata` and `partition` sections are considered only when `train` operation runs. In this case we ran the `profile` operation to extract features, so those values are ignored by deepprofiler.

You can find more info at https://cytomining.github.io/DeepProfiler-handbook/

**Comment by tahereT** [comment]
Hi @johnarevalo, thanks for your answer.
